### ISSUE TITLE
Optimize for readability and clarity, instead of performance

### ISSUE THREAD

**Issue by Tylersuard** [pytorch/pytorch#160640]
### 🚀 The feature, motivation and pitch

Currently PyTorch is optimized for speedy performance, which is great, but it does add some features that make things a little bit difficult for beginners. Adding tensors to devices and then detaching them, choosing an optimizer, zeroing out the gradients, the whole super().__init__() every time you need to create a new network, the complex structure of data loaders and datasets, etc. It would be great if there was an abstract wrapper for beginners, or just for people who want to do things quickly without having to take care of all the details.

Example:

```
from torch import simpletorch

neural_net = simpletorch.NeuralNet(
    inputs=100, 
    outputs=100, 
    layers=5)

training_data = simpletorch.image_data("./images")

neural_net.train(
    epochs=3, 
    training_data=training_data)

answer = neuralnet.guess("./test_images/test_image.jpg")

```

**Comment by Aidyn-A** [comment]
Well, for the purposes like that there are [scikit-learn](https://github.com/scikit-learn/scikit-learn), its CUDA accelerated analogue [cuML](https://github.com/rapidsai/cuml) and many other projects on GitHub. I doubt this is the path PyTorch is ever going to take.

**Comment by malfet** [comment]
Thank you for your suggestion. It feels like a great topic for discussion at https://discuss.pytorch.org/
